### PR TITLE
fix virus deleting after eaten

### PR DESCRIPTION
**Pull request by vikramdurai**
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/huytd/agar.io-clone/pull/493%23discussion_r112343089%22%2C%20%22https%3A//github.com/huytd/agar.io-clone/pull/493%23discussion_r112386555%22%2C%20%22https%3A//github.com/huytd/agar.io-clone/pull/493%23discussion_r112390252%22%2C%20%22https%3A//github.com/huytd/agar.io-clone/pull/493%23discussion_r112392097%22%2C%20%22https%3A//github.com/huytd/agar.io-clone/pull/493%23discussion_r112608079%22%2C%20%22https%3A//github.com/huytd/agar.io-clone/pull/493%23discussion_r112608080%22%2C%20%22https%3A//github.com/huytd/agar.io-clone/pull/493%23discussion_r112608137%22%2C%20%22https%3A//github.com/huytd/agar.io-clone/pull/493%23issuecomment-296276380%22%2C%20%22https%3A//github.com/huytd/agar.io-clone/pull/493%23issuecomment-296358338%22%5D%2C%20%22comments%22%3A%20%7B%22Pull%202e32e22e3edc55a937b07cf37d491d6e0b1a0ea1%20src/server/server.js%2017%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/huytd/agar.io-clone/pull/493%23discussion_r112343089%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22While%20you%27re%20at%20it%2C%20can%20you%20fix%20this%20check%3F%20It%20should%20be%20%60virusCollision.length%20%3E%200%60%22%2C%20%22created_at%22%3A%20%222017-04-20T00%3A24%3A47Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars1.githubusercontent.com/u/2415829%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/abalabahaha%22%7D%7D%2C%20%7B%22body%22%3A%20%22...%22%2C%20%22created_at%22%3A%20%222017-04-20T07%3A47%3A12Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars3.githubusercontent.com/u/7627362%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/vikramdurai%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222017-04-21T04%3A42%3A25Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars3.githubusercontent.com/u/7627362%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/vikramdurai%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20src/server/server.js%3AL571-579%22%7D%2C%20%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/huytd/agar.io-clone/pull/493%23issuecomment-296276380%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Nice%20try%20%5Cud83d%5Cudc4f%20keep%20going%21%20%40vikramdurai%20%22%2C%20%22created_at%22%3A%20%222017-04-21T18%3A55%3A46Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars2.githubusercontent.com/u/613943%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/huytd%22%7D%7D%2C%20%7B%22body%22%3A%20%22Ouch.%20I%20think%20my%20head%20hurts.%20Can%20we%20switch%20the%20server%20to%20anything%20_other_%20than%20Node%3F%20It%27s%20certainly%20worth%20considering.%22%2C%20%22created_at%22%3A%20%222017-04-22T08%3A43%3A19Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars3.githubusercontent.com/u/7627362%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/vikramdurai%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%20b66912481e86c5fb90356ebb548c76a4877c4b31%20src/server/server.js%2022%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/huytd/agar.io-clone/pull/493%23discussion_r112390252%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Expected%20an%20assignment%20or%20function%20call%20and%20instead%20saw%20an%20expression.%22%2C%20%22created_at%22%3A%20%222017-04-20T08%3A07%3A45Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars3.githubusercontent.com/u/6697940%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/houndci-bot%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-complete%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222017-04-21T04%3A42%3A26Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars3.githubusercontent.com/u/7627362%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/vikramdurai%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20src/server/server.js%3AL570-579%22%7D%2C%20%22Pull%20247d2fe5f75215cf7f524b6e0ece0330f418150a%20src/server/server.js%2020%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/huytd/agar.io-clone/pull/493%23discussion_r112392097%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Expected%20an%20identifier%20and%20instead%20saw%20%27%29%27.%3Cbr%3EExpected%20an%20assignment%20or%20function%20call%20and%20instead%20saw%20an%20expression.%3Cbr%3EMissing%20semicolon.%22%2C%20%22created_at%22%3A%20%222017-04-20T08%3A17%3A12Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars3.githubusercontent.com/u/6697940%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/houndci-bot%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%22%2C%20%22created_at%22%3A%20%222017-04-21T04%3A43%3A08Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars3.githubusercontent.com/u/7627362%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/vikramdurai%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20src/server/server.js%3AL570-579%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/huytd/agar.io-clone/pull/493#issuecomment-296276380'>General Comment</a></b>
- <a href='https://github.com/huytd'></a> Nice try 👏 keep going! @vikramdurai
- <a href='https://github.com/vikramdurai'></a> Ouch. I think my head hurts. Can we switch the server to anything _other_ than Node? It's certainly worth considering.
- [x] <a href='#crh-comment-Pull 2e32e22e3edc55a937b07cf37d491d6e0b1a0ea1 src/server/server.js 17'></a> &nbsp;<b><a href='https://github.com/huytd/agar.io-clone/pull/493#discussion_r112343089'>File: src/server/server.js:L571-579</a></b>
- <a href='https://github.com/abalabahaha'></a> While you're at it, can you fix this check? It should be `virusCollision.length > 0`
- <a href='https://github.com/vikramdurai'></a> ...
- [x] <a href='#crh-comment-Pull b66912481e86c5fb90356ebb548c76a4877c4b31 src/server/server.js 22'></a> &nbsp;<b><a href='https://github.com/huytd/agar.io-clone/pull/493#discussion_r112390252'>File: src/server/server.js:L570-579</a></b>
- <a href='https://github.com/houndci-bot'></a> Expected an assignment or function call and instead saw an expression.
- [x] <a href='#crh-comment-Pull 247d2fe5f75215cf7f524b6e0ece0330f418150a src/server/server.js 20'></a> &nbsp;<b><a href='https://github.com/huytd/agar.io-clone/pull/493#discussion_r112392097'>File: src/server/server.js:L570-579</a></b>
- <a href='https://github.com/houndci-bot'></a> Expected an identifier and instead saw ')'.<br>Expected an assignment or function call and instead saw an expression.<br>Missing semicolon.

<a href='https://www.codereviewhub.com/huytd/agar.io-clone/pull/493?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/huytd/agar.io-clone/pull/493?approve=1'></a>&nbsp;<a href='https://github.com/huytd/agar.io-clone/pull/493'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by huytd** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/huytd/agar.io-clone/pull/493'>Refresh</a>.<a href='#crh-update-none'></a>

File: src/server/server.js [hunk 1]
```diff
@@ -566,7 +571,8 @@ function tickPlayer(currentPlayer) {
            .reduce( function(a, b, c) { return b ? a.concat(c) : a; }, []);
 
         if(virusCollision > 0 && currentCell.mass > virus[virusCollision].mass) {
```
Comment (abalabahaha) [inline src/server/server.js]: While you're at it, can you fix this check? It should be `virusCollision.length > 0`

Comment (vikramdurai) [inline src/server/server.js]: ...

Comment (vikramdurai) [inline src/server/server.js]: :+1:<a href='#crh-update-none'></a>

File: src/server/server.js [hunk 2]
```diff
@@ -565,8 +570,9 @@ function tickPlayer(currentPlayer) {
         var virusCollision = virus.map(funcFood)
            .reduce( function(a, b, c) { return b ? a.concat(c) : a; }, []);
 
-        if(virusCollision > 0 && currentCell.mass > virus[virusCollision].mass) {
-          sockets[currentPlayer.id].emit('virusSplit', z);
+        if(virusCollision.length > 0 && currentCell.mass > virus[virusCollision].mass) {
+            sockets[currentPlayer.id].emit('virusSplit', z);
+ +          virusCollision.forEach(deleteVirus);
```
Comment (vikramdurai) [inline src/server/server.js]: :+1:<a href='#crh-update-complete'></a>

File: src/server/server.js [hunk 3]
```diff
@@ -565,8 +570,9 @@ function tickPlayer(currentPlayer) {
         var virusCollision = virus.map(funcFood)
            .reduce( function(a, b, c) { return b ? a.concat(c) : a; }, []);
 
-        if(virusCollision > 0 && currentCell.mass > virus[virusCollision].mass) {
-          sockets[currentPlayer.id].emit('virusSplit', z);
+        if(virusCollision > 0 && currentCell.mass > virus[virusCollision].mass) {) {
```
Comment (vikramdurai) [inline src/server/server.js]: :+1:

**Comment by huytd** [comment]
Nice try 👏 keep going! @vikramdurai

**Comment by vikramdurai** [comment]
Ouch. I think my head hurts. Can we switch the server to anything _other_ than Node? It's certainly worth considering.

**Comment by abalabahaha** [comment]
PR was kinda broken so I reimplemented it... Thank you for your contribution!

**Comment by vikramdurai** [comment]
:+1:

**Pull request closed without merging**